### ISSUE TITLE
Ranged expert and dead eye not functioning with thrown weapons

### ISSUE THREAD

**Issue by Valensfd37** [SolastaMods/SolastaUnfinishedBusiness#2718]
Hi ! I've come to the discovery that Ranged expert and Dead Eye do not seem to recognize thrown weapons for half of their features.

I'll explain my discovery process : 
First up I was levelling an armorer Inventor with the Infiltrator armor type, got the dead eye feat for them and realized it didn't work, after looking around here found that it's a known bug, so I redid the level up and took ranged expert. Now I saw that it didn't work either, I assumed it might have been because it wasn't a "one-handed" weapon per say, so I had my inventor holding a dagger to throw... Didn't work either, only way to get the bonus action to show up is by wielding two daggers but then it's just TWF activating.

In case it was some weird inventor interaction, I went up and made two new character specifically to try it, a lv1 Fighter/lv4 Inventor human (to get the proficiency with hand crossbow) and a lv4 Fighter human, both having dead eye and ranged expert

Through testing I found that : 
- While wielding a hand crossbow, Ranged expert function perfectly with the artificer (Any first attack, even the lighting launcher, allows for the hand crossbow bonus action) but Dead eye attack debuff and damage bonus only appeared on the hand crossbow shots.
- While wielding Thrown weapons, even on the full fighter nothing triggered : Using Two javelin (so TWF wouldn't interfere)  the bonus action never appeared nor did the attack or damage change with dead eye.
- I also tested the other features of both feat : Lighting launcher and the thrown weapon ignore the engaged in melee disadvantage, the covers AC bonus and long range disadvantages, as they should.
- I also tried swapping load-outs in between action and got the same results.

In conclusion:
- Ranged expert bonus action attack doesn't activate while wielding a thrown weapon, despite being in the description
- Dead eye Attack debuff and bonus doesn't activate with ranged weapon (this might be on purpose though, as it doesn't specify thrown weapon) but its other features do.

Thank you for your time!

**Comment by Valensfd37** [comment]
Posting didn't seem to work with the save file included, so i'll try here : 
[RangedExpertThrownWeapon.zip](https://github.com/SolastaMods/SolastaUnfinishedBusiness/files/10957754/RangedExpertThrownWeapon.zip)

**Comment by ThyWoof** [comment]
Thanks. It’s pretty hard to add thrown functionality to both so will fix ranger mastery description.
